### PR TITLE
Add components namespace and id scopes

### LINKED ISSUE #1045: Policy and Governing Components
I just wanted to open a discussion to understand the current thinking about applying policies and governance to Dapr components in K8s. Right now, in K8s, AFAIK once a component definition is deployed it is open to be consumed by any Dapr instance deployed to the cluster as per the [loader](https://github.com/dapr/dapr/blob/594af6948064d26dc5df8da68b310884948bf806/pkg/components/kubernetes_loader.go#L35). Loading all components to every daprd instance is clearly not going to work in an actual use case. Do we already have a plan for addressing this or do we need to discuss it further?

### LINKED ISSUE #1181: Scope components to namespace and app id
Currently, when daprd is injected into a pod it will look for dapr components across all namespaces. Instead, from a security perspective, it should be scoped to only load components from the namespace in which it was injected. 

Also see #1045

### PR DESCRIPTION
**Pull request by yaron2**
This PR adds important security capabilities, namely allowing users to scope components for individual apps, and disallowing a sidecar to load components from a namespace it isn't deployed in.

Closes #1181 
Closes #1045 

This PR also adds a boilerplate header for CRD generation.

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -140,11 +141,16 @@ func (a *DaprRuntime) Run(opts ...Option) error {
 	return nil
 }
 
+func (a *DaprRuntime) getNamespce() string {
```
Comment (youngbupark) [inline pkg/runtime/runtime.go]: typo. getNamespace

File: pkg/runtime/runtime_test.go [hunk 2]
```diff
@@ -610,3 +610,110 @@ func TestReadInputBindings(t *testing.T) {
...
+
+		assert.Empty(t, ns)
+	})
+
+	t.Run("non-empty namespace", func(t *testing.T) {
+		os.Setenv("NAMESPACE", "a")
+		defer os.Clearenv()
+
+		rt := NewTestDaprRuntime(modes.StandaloneMode)
+		ns := rt.getNamespce()
```
Comment (youngbupark) [inline pkg/runtime/runtime_test.go]: typo getNameSpace()

**Comment by yaron2** [comment]
@youngbupark recheck plz

File: tools/boilerplate.go.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*
+Copyright The Dapr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (youngbupark) [inline tools/boilerplate.go.txt]: K8s is under apache 2?

Comment (yaron2) [reply tools/boilerplate.go.txt]: yeah

**Pull request merged**